scene_file: "empty.ttt"
objects: [
  {
    model_filename: "ros_clock.ttm"
  },
  {
    model_filename: "stl_object.ttm",
    stl_filename: "surface.stl",
    position: [0.0, 0.0, 0.0],
    orientation: [0.0, 0.0, 0.0, 1.0],
  },
]
robots: [
  {
    model_filename: "ros_zumy_base.ttm",
    position: [0.0, 0.0, 0.0],
    orientation: [0.0, 0.0, 0.0, 1.0],
    children: [
      {
        model_filename: "ros_camera.ttm",
        topic: "camera",
        position: [0.0, 0.0, 0.05],
        orientation: [-0.5, 0.5, -0.5, 0.5],
      },
      {
        model_filename: "ros_scanner.ttm",
        topic: "scanner",
        position: [0.0, 0.01, 0.05],
        orientation: [-0.5, 0.5, -0.5, 0.5],
      },
      {
        model_filename: "ros_imu.ttm",
        topic: "imu",
        position: [0.0, 0.0, 0.02],
        orientation: [0.0, 0.0, 0.0, 1.0],
      },
      {
        model_filename: "ros_markers.ttm",
        topic: "markers",
        position: [0.0, 0.0, 0.0],
        orientation: [0.0, 0.0, 0.0, 1.0],
        marker_positions: [
          {h:   0.0, x: -0.100, y:  0.060, z: 0.060},
          {h:  30.0, x: -0.060, y: -0.030, z: 0.100},
          {h:  60.0, x:  0.000, y: -0.075, z: 0.060},
          {h:  90.0, x:  0.060, y: -0.030, z: 0.100},
          {h: 120.0, x:  0.100, y:  0.060, z: 0.060},
          {h: 150.0, x:  0.000, y:  0.030, z: 0.100},
        ]
      },
    ]
  },
  {
    model_filename: "ros_zumy_base.ttm",
    position: [0.5, 0.2, 0.0],
    orientation: [0.0, 0.0, 0.0, 1.0],
    children: [
      {
        model_filename: "ros_camera.ttm",
        topic: "camera",
        position: [0.0, 0.0, 0.05],
        orientation: [-0.5, 0.5, -0.5, 0.5],
      },
      {
        model_filename: "ros_scanner.ttm",
        topic: "scanner",
        position: [0.0, 0.01, 0.05],
        orientation: [-0.5, 0.5, -0.5, 0.5],
      },
      {
        model_filename: "ros_imu.ttm",
        topic: "imu",
        position: [0.0, 0.0, 0.02],
        orientation: [0.0, 0.0, 0.0, 1.0],
      },
      {
        model_filename: "ros_markers.ttm",
        topic: "markers",
        position: [0.0, 0.0, 0.0],
        orientation: [0.0, 0.0, 0.0, 1.0],
        marker_positions: [
          {h:   0.0, x:  0.000, y: -0.075, z: 0.060},
          {h:  30.0, x:  0.100, y:  0.060, z: 0.060},
          {h:  60.0, x:  0.060, y: -0.030, z: 0.100},
          {h:  90.0, x: -0.060, y:  0.030, z: 0.100},
          {h: 120.0, x: -0.100, y: -0.060, z: 0.060},
          {h: 150.0, x:  0.000, y:  0.075, z: 0.070},
        ]
      },
    ]
  },
]
